### PR TITLE
Return an user-friendly message on TPM permission errors

### PR DESCRIPTION
**Pull request by codingllama**
Transform this:

```shell
$ tsh device enroll --current-device
> ERROR: opening tpm
> open /dev/tpmrm0: permission denied
```

into this:

```shell
$ tsh device enroll --current-device
> ERROR: Failed to open the TPM device. Consider assigning the user to the `tss`
group or creating equivalent udev rules.See
https://goteleport.com/docs/access-controls/device-trust/device-management/#troubleshooting.
```

(I've also snuck in a couple of small change along with it.)

https://github.com/gravitational/teleport.e/issues/827

### CODE REVIEW

File: lib/devicetrust/native/device_linux.go [hunk 1]
```diff
@@ -51,34 +51,64 @@ var linuxDevice = &tpmDevice{
...
+		return err
+	}
+	log.
+		WithError(err).
+		Debug("TPM: Replacing TPM permission error with a more friendly one")
+
+	return errors.New("" +
+		"Failed to open the TPM device. " +
+		"Consider assigning the user to the `tss` group or creating equivalent udev rules. " +
+		"See https://goteleport.com/docs/access-controls/device-trust/device-management/#troubleshooting.")
```
Comment (codingllama) [inline lib/devicetrust/native/device_linux.go:109]: Note: I'll update this section with instructions shortly.

Comment (Tener) [reply lib/devicetrust/native/device_linux.go:109]: Is the `tss` group a standard thing? Should we check the owner of the device to make sure?

**Review by Tener — APPROVED** [review]

File: lib/devicetrust/native/device_linux.go [hunk 2]
```diff
@@ -51,34 +51,64 @@ var linuxDevice = &tpmDevice{
...
 }
 
+func rewriteTPMPermissionError(err error) error {
+	// We are looking for an error that looks roughly like this:
+	//
+	// 	err = &fs.PathError{
+	// 		Path: "/dev/tpmrm0",
+	// 		Err: fs.ErrPermission,
+	// 	}
+	if !errors.Is(err, fs.ErrPermission) {
```
Comment (zmb3) [inline lib/devicetrust/native/device_linux.go:94]: Won't the errors.As check below make this check redundant?

**Review by zmb3 — APPROVED** [review]

File: lib/devicetrust/native/device_linux.go (hunk 1 above)
Comment (codingllama) [reply lib/devicetrust/native/device_linux.go:109]: As far as I can tell, yes, it exists in my Ubuntu install by default. I believe it gets created along with TPM tools: https://github.com/tpm2-software/tpm2-tss/tree/master#overview.

Not sure what you meant by checking the device owner?

File: lib/devicetrust/native/device_linux.go (hunk 2 above)
Comment (codingllama) [reply lib/devicetrust/native/device_linux.go:94]: I suppose we could see PathErrors with other underlying reasons, so I was a bit overzealous here.

File: lib/devicetrust/native/device_linux.go (hunk 1 above)
Comment (codingllama) [reply lib/devicetrust/native/device_linux.go:109]: A better link about tss: https://github.com/tpm2-software/tpm2-tss/blob/master/dist/tpm-udev.rules#L4.

**Pull request merged**